evergreen/compile.sh

```diff
@@ -9,5 +9,15 @@ set -o errexit  # Exit the script with error if any of the commands fail
 
 echo "Compiling .NET driver"
 
-for var in TMP TEMP NUGET_PACKAGES NUGET_HTTP_CACHE_PATH APPDATA; do setx $var z:\\data\\tmp; export $var=z:\\data\\tmp; done
-powershell.exe .\\build.ps1 -target Build -Verbosity Diagnostic
+if [ "$OSTYPE" = "cygwin" ]; then
```

Comment:
because of this comment https://github.com/durran/mongo-csharp-driver/pull/2#discussion_r526987531, should we use just `$OS`?

Reply:
Done.